extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusSecurityIdentity.java

```diff
@@ -121,11 +123,13 @@ public static Builder builder() {
     }
 
     public static Builder builder(SecurityIdentity identity) {
-        return new Builder()
+        Builder builder = new Builder()
                 .addAttributes(identity.getAttributes())
                 .addCredentials(identity.getCredentials())
                 .addRoles(identity.getRoles())
                 .setPrincipal(identity.getPrincipal());
+        builder.anonymous = identity.isAnonymous();
```

Comment:
I would continue using the fluent API here

Reply:
@renegrob oh yes, that line has stayed there before I added `setAnonymous`, good catch